src/urllib3/connectionpool.py

```diff
@@ -489,7 +489,7 @@ def _make_request(
                 new_e, (OSError, NewConnectionError, TimeoutError, SSLError)
             ) and (conn and conn.proxy and not conn.has_connected_to_proxy):
                 new_e = _wrap_proxy_error(new_e, conn.proxy.scheme)
-            raise new_e
+            raise new_e from e
```

Comment:
`_wrap_proxy_error` does already explicitly set the `__cause__` and `raise new_e from e`  will overwrite that `__cause__`

so in the case of e being a `CertificateError` the current code can produce ProxyError -> SSLError -> CertificateError but with this change it will produce ProxyError -> CertificateError instead. 

So I think in this case if we really would like to get the "The above exception was the direct cause of the following exception:" instead of "During handling of the above exception, another exception occurred:"  would be better to set `new_e.__cause__` explicitly after the `SSLError`.


Reference https://docs.python.org/3/library/exceptions.html#exception-context